contracts/modules/grouping/GroupingModule.sol

```diff
@@ -0,0 +1,220 @@
+        // initialize royalty vault
+        // transfer all royalty tokens to the  group pool
+        // transfer Group NFT to msg.sender
+        GROUP_NFT.safeTransferFrom(address(this), msg.sender, groupNftId);
+        emit IPGroupRegistered(groupId, groupPool);
+    }
+
+    /// @notice Whitelists a group reward pool.
+    /// @param rewardPool The address of the group reward pool.
+    function whitelistGroupRewardPool(address rewardPool) external restricted {
+        GROUP_IP_ASSET_REGISTRY.whitelistGroupRewardPool(rewardPool);
+    }
+
+    /// @notice Adds IP to group.
+    /// the function must be called by the Group IP owner or an authorized operator.
+    /// @param groupIpId The address of the group IP.
+    /// @param ipIds The IP IDs.
+    function addIp(address groupIpId, address[] calldata ipIds) external whenNotPaused verifyPermission(groupIpId) {
+        if (LICENSE_REGISTRY.hasDerivativeIps(groupIpId)) {
+            revert Errors.GroupingModule__GroupIPHasDerivativeIps(groupIpId);
+        }
+
+        // check if the group IP has license terms and minting fee is 0
+        if (LICENSE_REGISTRY.getAttachedLicenseTermsCount(groupIpId) == 0) {
+            revert Errors.GroupingModule__GroupIPHasNoLicenseTerms(groupIpId);
+        }
+        (address groupLicenseTemplate, uint256 groupLicenseTermsId) = LICENSE_REGISTRY.getAttachedLicenseTerms(
+            groupIpId,
+            0
+        );
+        PILTerms memory groupLicenseTerms = IPILicenseTemplate(groupLicenseTemplate).getLicenseTerms(
+            groupLicenseTermsId
+        );
+        if (groupLicenseTerms.defaultMintingFee != 0) {
+            revert Errors.GroupingModule__GroupIPHasMintingFee(groupIpId, groupLicenseTemplate, groupLicenseTermsId);
+        }
+
+        GROUP_IP_ASSET_REGISTRY.addGroupMember(groupIpId, ipIds);
+        IGroupRewardPool pool = IGroupRewardPool(GROUP_IP_ASSET_REGISTRY.getGroupRewardPool(groupIpId));
+        for (uint256 i = 0; i < ipIds.length; i++) {
+            // check if the IP has the same license terms as the group
+            if (!LICENSE_REGISTRY.hasIpAttachedLicenseTerms(ipIds[i], groupLicenseTemplate, groupLicenseTermsId)) {
+                revert Errors.GroupingModule__IpHasNoGroupLicenseTerms(
+                    ipIds[i],
+                    groupLicenseTemplate,
+                    groupLicenseTermsId
+                );
+            }
+            pool.addIp(groupIpId, ipIds[i]);
+        }
+
+        emit AddedIpToGroup(groupIpId, ipIds);
+    }
+
+    /// @notice Removes IP from group.
+    /// the function must be called by the Group IP owner or an authorized operator.
+    /// @param groupIpId The address of the group IP.
+    /// @param ipIds The IP IDs.
+    function removeIp(address groupIpId, address[] calldata ipIds) external whenNotPaused verifyPermission(groupIpId) {
+        if (LICENSE_REGISTRY.hasDerivativeIps(groupIpId)) {
```

Comment:
There could be a front-running risk here. Group owner removes the IPs after having sold a license at a high price. 

The addIp and removeIp should revert if the group has minted a license token - the same happens for linking at the moment

Reply:
Add restriction that lock the group when it has minted any license token.